homeassistant/components/atag/config_flow.py

```diff
@@ -0,0 +1,59 @@
+"""Config flow for the Atag component."""
+from aiohttp import ClientSession
+from pyatag import DEFAULT_PORT, AtagDataStore, AtagException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE, CONF_EMAIL, CONF_HOST, CONF_PORT
+from homeassistant.core import callback
+
+from . import DOMAIN  # pylint: disable=unused-import
+
+DATA_SCHEMA = {
+    vol.Required(CONF_HOST): str,
+    vol.Optional(CONF_EMAIL): str,
+    vol.Required(CONF_PORT, default=DEFAULT_PORT): vol.All(int, vol.Range(min=0)),
```

Comment:
I would suggest to use `vol.Coerce(int)` since it is used by the config flow of mqtt and adguard.
```suggestion
    vol.Required(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
```

Reply:
Changed to Coerce, note that this opens up the potential of negative ports being entered.